view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js

```diff
@@ -373,18 +390,31 @@ define(
             placeRedirectOrder: function(data) {
                 // Place Order but use our own redirect url after
                 var self = this;
+                fullScreenLoader.startLoader();
 
                 var messageContainer = this.messageContainer;
                 if(brandCode()) {
                     messageContainer = self.messageComponents['messages-' + brandCode()];
                 }
-                
+
+                $('.hpp-message').slideUp();
+
                 this.isPlaceOrderActionAllowed(false);
-                fullScreenLoader.startLoader();
                 $.when(
                     placeOrderAction(data, messageContainer)
                 ).fail(
-                    function () {
+                    function (response) {
+                        fullScreenLoader.stopLoader();
+                        //messageContainer.addErrorMessage((response['responseJSON'].message).replace('%1', response['responseJSON'].parameters[0]));
+                        if (!!response['responseJSON'].parameters) {
+                            $("#messages-" + brandCode()).text((response['responseJSON'].message).replace('%1', response['responseJSON'].parameters[0])).slideDown();
+                        } else {
+                            $("#messages-" + brandCode()).text(response['responseJSON'].message).slideDown();
+                        }
+
```

Comment:
The Magento default backend validator is sending back the validation errors e.g when we don't have billing agreement filled in. This can be a case in guest checkout with virtual product. These error messages are passed back by default from Magento and would have been shown already if the default error message showing method worked (`messageContainer.addErrorMessage`)
I could also show a general error message like - "make sure you filled in all the required fileds" but I think it helps the customers more.